blog/2023-03-20-handler-slack-commands/index.md

```diff
@@ -0,0 +1,230 @@
+}
+
+function getHelp() {
+  const help = `Supported commands
+  help - prints this command
+  echo - prints input
+  `;
+  return help;
+}
+```
+
+With this example, you have the keys to build complex Slackbots from Windmill. In the following section, we'll see how to handle more parameters, and use approval steps to go back and forth between Slack and Windmill.
+
+## Enhance your Slackbot with more parameters, resources and approval steps
+
+The goal of the current section is to give you more hints on the potentiality of the Slackbot. Below is a flow that was built for more custom needs. We kept it simple, but we can't even think of all the possibilities of slackbots built with Windmill.
+
+<!--(but maybe our users have more ideas) + link to Atelier post, when published-->
+
+![Slack commands hander, 2nd example](./3-commands-handler-n2.png "Second example of a Slack commands handler")
+
+### Fine-tuning the parser to manage longer commands
+
+The default parser of the first example has the advantage of being simple as it parses word by word. Since we want to give more comprehensive parameters (sentences etc.), I chose to manage the parameters following the command separated with `" "` instead of spaces, as shown with the following code (generated with Open AI, remember I'm not a developer):
+
+```js
+export async function main(text_input: string): Output {
+  let tokenized: string[] = [];
+  if (text_input) {
+    let start = 0;
+    let inString = false;
+    for (let i = 0; i < text_input.length; i++) {
+      if (text_input[i] === '"') {
+        inString = !inString;
+      }
+      if (text_input[i] === ' ' && !inString) {
+        tokenized.push(text_input.slice(start, i).replace(/"/g, ''));
+        start = i + 1;
+      }
+    }
+    tokenized.push(text_input.slice(start).replace(/"/g, ''));
+  }
+  const command = tokenized[0] || 'help';
+  const input = tokenized[1];
+  const parameters1 = tokenized[2];
+  const parameters2 = tokenized[3];
+
+  return { command, input, parameters1, parameters2 };
+}
+```
+
+With the following parser, if from Slack I write `/windmill coolcommand "the 1st sentence" "the 2nd sentence" "the 3rd sentence"`, it will result in:
+
+```js
+{
+    "input": "the 1st sentence",
+    "command": "coolcommand",
+    "parameters1": "the 2nd sentence",
+    "parameters2": "the 3rd sentence"
+}
+```
+
+### Interacting with resources
+
+<video
+    className="border-2 rounded-xl object-cover w-full h-full"
+    autoPlay
+    loop
+    controls
+    id="main-video"
+    src="/videos/command_event.mp4"
+    alt="container component"
+/>
+
+<br/>
+
+All commands (except the default one) of this slackbot deal with [resources](https://docs.windmill.dev/docs/core_concepts/resources_and_types). Adding resources is easy on Windmill: a lot of resources are [already available](https://hub.windmill.dev/resources), or you can [create your own](https://docs.windmill.dev/docs/core_concepts/resources_and_types#create-a-resource).
```

Comment:
Many resource types

important to be clear, resource is the value, resource type is its type